target/scripts/start-mailserver.sh

```diff
@@ -18,7 +18,7 @@ _obtain_hostname_and_domainname
 # These variables must be defined first; They are used as default values for other variables.
 VARS[POSTMASTER_ADDRESS]="${POSTMASTER_ADDRESS:=postmaster@${DOMAINNAME}}"
 VARS[REPORT_RECIPIENT]="${REPORT_RECIPIENT:=${POSTMASTER_ADDRESS}}"
-VARS[REPORT_SENDER]="${REPORT_SENDER:=mailserver-report@${DOMAINNAME}}"
+VARS[REPORT_SENDER]="${REPORT_SENDER:=mailserver-report@$(hostname -f)}"
```

Comment:
I remember this was `HOSTNAME` before. I think there also is a test you'll need to adjust.

Reply:
You are right, one test failed. I must have done something wrong yesterday, as my local tests passed successful 😕

Reply:
I see, but here in `start-mailserver.sh`, we could also use the same env variable, right?

Reply:
I re-read your comment 😆 `$HOSTNAME` should also be fine.

PS; I still find it very confusing, that we modify system variables. `$HOSTNAME` is set by docker.
If I `bash` into DMS and `echo $HOSTNAME`, just `mail` is returned,  while in the `start-mailserver.sh` context `$HOSTNAME` is a FQDN:

```bash
root@mail:/# echo $HOSTNAME
mail
root@mail:/# source ./usr/local/bin/helpers/dns.sh
root@mail:/# _obtain_hostname_and_domainname
root@mail:/# echo $HOSTNAME
mail.chill.to
```

Maybe in a future PR we should change that. Instead of modifying $HOSTNAME, we should use $DMS_HOSTNAME or something else to make the difference clear and avoid confusion.

Reply:
I re-read your comment 😆 `$HOSTNAME` should also be fine.

PS; I find it very confusing, that we modify system variables. This is bad practice IMO. `$HOSTNAME` is set by docker.
If I `bash` into DMS and `echo $HOSTNAME`, just `mail` is returned,  while in the `start-mailserver.sh` context `$HOSTNAME` is a FQDN:

```bash
root@mail:/# echo $HOSTNAME
mail
root@mail:/# source ./usr/local/bin/helpers/dns.sh
root@mail:/# _obtain_hostname_and_domainname
root@mail:/# echo $HOSTNAME
mail.example.com
```

Maybe in a future PR we should change that. Instead of modifying $HOSTNAME, we should use $DMS_HOSTNAME or something else to make the difference clear and avoid confusion.

Reply:
> I re-read your comment 😆 $HOSTNAME should also be fine.

Just tested it - it's not. Because the containers `$HOSTNAME` (no FQDN) is different to `$HOSTNAME` (FQDN) used in `start-mailserver.sh`.

Reply:
I agree, this needs some reworking. I think @polarathene has this in his backlog somewhere :D

Reply:
> PS; I find it very confusing, that we modify system variables. This is bad practice IMO. $HOSTNAME is set by docker.

:+1: 

I've already expressed this concern in the past, and there was talk to change to `DMS_HOSTNAME` / `DMS_FQDN` IIRC, but required tests to be adapted for running through some different configs as I believe dropping `DOMAINNAME` was also being considered.